File: java/PalindromeNumber.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0 || (x!=0 && x%10==0)) return false;
        int tmp = 0;
        while (x > tmp) {
            tmp = tmp*10 + x%10;
            x /= 10;
        }
        return (x == tmp || tmp/10 == x)? true:false;
    }
}

File: cpp/palindromenumber.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x<0 || (x != 0 and x%10 == 0)) return false;
        int num = 0;
        while (num < x) {
            num = num*10+x%10;
            x /= 10;
        }
        if (num == x or num/10 == x) return true;
        return false;
    }
};
